### DISCUSSION TITLE
OPEN DTU  for Hoymiles   ZERO  Export with Shelly EM and Shelly Smart Plug

### DISCUSSION THREAD

**Discussion by Stratis63b** [tbnobody/OpenDTU#800] [General]
The reason i ma writing to the Group is to see if you have tested the product with Shelly EM and smart plug 
At the moment Hoymiles proposes  DTU PRO and Chint 666 3ct , a very expensive solution for Balcony Solar modules, 
So OPEN DTU solves the DTU PRO issue but still we need th Chint Smart meter .
Sorry for my English but my German is worst
Hallo Lukas
und herzlichen Glückwunsch zu Ihrem OPEN-DTU-Projekt,
Der Grund, warum ich Ihnen schreibe, ist, um zu sehen, ob Sie das Produkt mit Shelly EM und Smart Plug getestet haben
Im Moment bietet Hoymiles DTU PRO und Chint 666 3ct an, eine sehr teure Lösung für Balkon-Solarmodule,
OPEn DTU löst also das DTU PRO-Problem, aber wir brauchen immer noch den Chint Smart Meter.
Sorry für mein Englisch, aber mein Deutsch ist am schlechtesten
Stratis

**Comment by mark82436** [comment]
Hallo Stratis,

ich habe bei mir openDTU mit Shelly 3EM und Shelly Plug ohne Probleme am laufen.
Als Nulleinspeisung Script benutze ich https://github.com/reserve85/HoymilesZeroExport/
Am Shelly Plug habe ich aber Tasmota installiert da man es besser Kalibrieren kann.

**Reply by FlechtkerH** [reply]
Der eingefügte Link funktioniert bei mir nicht. Könntest du da nochmal drauf schauen?

**Reply by mark82436** [reply]
habe es geändert

**Comment by Stratis63b** [comment]
I guess you meant this
https://github.com/reserve85/HoymilesZeroExport

So you got Ahoy DTU  or Open DTU and then you added this script, Cause the
installation instructions are not very clear.

On Sun, Apr 16, 2023 at 8:35 AM mark82436 ***@***.***> wrote:

> Hallo Stratis,
>
> ich habe bei mir openDTU mit Shelly 3EM und Shelly Plug ohne Probleme am
> laufen.
> Als Nulleinspeisung Script benutze ich
> https://github.com/reserve85/HoymilesZeroExport/ <http://url>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/tbnobody/OpenDTU/discussions/800#discussioncomment-5626696>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A4RQFPQHBYEXHHKLT3EIQULXBOAIHANCNFSM6AAAAAAW3MMRPE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

*Stratis Telloglou*

*OPERATIONS – SENIOR PROJECT MANAGEMENT*

* +30 2118004039 |* *)** +30 6972004721|*

*** ***@***.*** ***@***.***>* |* Stratis63 |
br.linkedin.com/in/stratistelloglou
<http://br.linkedin.com/in/stratistelloglou>*

**Reply by mark82436** [reply]
Yes i have runing this script on linux client and a openDTU  with HM-1500
Gridmeter are Shelly 3EM and Solarmeter is Shelly Plug with Tasmota Software.

The Shelly Plug is optional the script are runing with gridmeter and dtu solarmeter are optional

**Comment by Stratis63b** [comment]
Can you give me some details on how you did the whole setup after you
installed the open dtu and how you connected the Shelly em

**Reply by Brutus0001** [reply]
Hallo, würde auch gern mehr darüber Wissen (kleine Anleitung) vielen Dank

**Reply by bscholtes** [reply]
for a first setup I prepared a docker container with suitable ini setup and entrypoint for a simple docker run ... 
with "--init" for SIGNAL handling and "--restart unless-stopped" for automatic restart to docker options 

Dockerbuild:
FROM python
ADD HoymilesZeroExport.py .
ADD HoymilesZeroExport_Config.ini .
RUN pip install --upgrade pip && pip install requests
HEALTHCHECK --interval=60s --timeout=2s --start-period=45s CMD [ "$(tail -1 /log/log | awk -F':' '{print $$1 ":" $$2}')" = "$(date '+%Y-%m-%d %H:%M')" ] || sh -c 'kill -s 15 1 && (sleep 10; kill -s 9 1)'
ENTRYPOINT ["python","./HoymilesZeroExport.py"]

TASMOTA TTL reader setup (like bitshake ESP8266 with HOLLEY e-meter):
USE_AHOY = true
USE_TASMOTA = true
AHOY_IP = 192.168.xxx.xxy
TASMOTA_IP = 192.168.xxx.xxx
TASMOTA_JSON_STATUS = StatusSNS
TASMOTA_JSON_PAYLOAD_MQTT_PREFIX = DTZ541
TASMOTA_JSON_POWER_MQTT_LABEL = curr_w
TASMOTA_JSON_POWER_CALCULATE = FALSE
TASMOTA_JSON_POWER_INPUT_MQTT_LABEL =
TASMOTA_JSON_POWER_OUTPUT_MQTT_LABEL =
USE_TASMOTA_INTERMEDIATE = false
